src/index.js

```diff
@@ -370,17 +370,22 @@ function Datastore(options) {
 
   this.clients_ = new Map();
   this.datastore = this;
+
   /**
    * @name Datastore#namespace
    * @type {string}
    */
   this.namespace = options.namespace;
+
+  const userProvidedProjectId =
+    options.projectId || process.env.DATASTORE_PROJECT_ID;
```

Comment:
This is flipped from how it was before, which I think is what would be expected. If the user manually provides a projectId, I think we're supposed to honor that and not do any magic. What do you think?

Reply:
My understanding is that environment variable should always be prioritized over anything written in the code. In my world, command line option > environment variable > local configuration file > global configuration file > default value in the source code. You might have different opinion though :)

Reply:
I went back to an [old issue]((https://github.com/GoogleCloudPlatform/google-cloud-node/issues/191) where we discussed a similar topic, and it reminded me of a feature we offer in our libraries... the user can uniquely authenticate multiple instantiations. So, I can hold an instance of `Datastore` for "project-a" and another for "project-b". We let the user choose at run-time which project each invocation refers to.

> command line option > environment variable > local configuration file > global configuration file > default value in the source code

I would agree with that, however, I believe you've ranked `options.projectId` as the "local configuration file". To me, the user explicitly writing "projectId: 'my-project'" is the same rank as a command line option. 

I think that if the user typed out "my-project", but we ignore that and use an env var, that would be unexpected for them.

Reply:
OK, let's leave it this way (probably until someone complains :) )